src/transformers/generation_utils.py

```diff
@@ -628,32 +646,29 @@ def _get_decoder_start_token_id(self, decoder_start_token_id: int = None, bos_to
 
     @staticmethod
     def _expand_inputs_for_generation(
-        input_ids: torch.LongTensor,
```

Comment:
This is now optional to share this beam search logic with contrastive search: in contrastive search, we don't need to replicate `input_ids`